### ISSUE TITLE
[Regression]: VSCode extension closes browser after test

### ISSUE THREAD

**Issue by pietdevries94** [microsoft/playwright#30779]
### Last Good Version

1.0.22

### First Bad Version

1.1.1

### Steps to reproduce

1. Enable show browser
2. Run a test

### Expected behavior

The browser stays open after running the test.

### Actual behavior

The test runs, but it closes the browser right away.

### Additional context

This change was not mentioned in the changelog. If this is deliberate, please add an option to keep the browser open.

The environment given here is mine, but my colleague has the same issue on Windows.

### Environment

```shell
System:
    OS: Linux 6.1 NixOS 23.11 (Tapir) 23.11 (Tapir)
    CPU: (16) x64 AMD Ryzen 9 5900HX with Radeon Graphics
    Memory: 11.29 GB / 27.27 GB
    Container: Yes
  Binaries:
    Node: 20.12.2 - /nix/store/71r8n7ckcpvav9qwshlr12hjd5nlchds-nodejs-20.12.2/bin/node
    npm: 10.5.0 - /nix/store/71r8n7ckcpvav9qwshlr12hjd5nlchds-nodejs-20.12.2/bin/npm
    pnpm: 8.15.5 - /nix/store/16s3dlkk4n6h9nh6mfp77fcm6gll0m5l-pnpm-8.15.5/bin/pnpm
  IDEs:
    VSCode: 1.89.1 - /etc/profiles/per-user/piet/bin/code
  Languages:
    Bash: 5.2.26 - /nix/store/h3bhzvz9ipglcybbcvkxvm4vg9lwvqg4-bash-5.2p26/bin/bash
  npmPackages:
    @playwright/test: ^1.44.0 => 1.44.0
```

**Comment by mxschmitt** [comment]
If "Show browser" is selected, the browser should stay open. I'm not able to reproduce - are you able to share your project with us? Does it happen on a new project created via "npm init playwright my-new-project"?

**Comment by pietdevries94** [comment]
In a new project it doesn't happen. However, I found the setting that breaks it. 

It breaks when you turn on video in the playwright config. We had it set to `on-first-retry`, but also breaks with `on`. We checked this both on my machine, as the Windows machine.

Because we only recorded on retries only, and we only retry on CI, we updated the config value to `process.env.CI ? 'on-first-retry' : 'off'` and this workaround works
